emperor/support_files/js/color-view-controller.js

```diff
@@ -56,62 +55,108 @@ define([
     var SLICK_WIDTH = 25, scope = this;
     var name, value, colorItem;
 
+    // Create scale div and checkbox for whether using scalar data or not
+    this.$scaleDiv = $('<div></div>')
+    this.$colorScale = $("<svg width='90%' height='40'></svg>");
+    this.$scaleDiv.append(this.$colorScale);
+    this.$scaled = $("<input type='checkbox' id='scaled'>");
```

Comment:
Would it make sense to have a plot-specific prefix that's a randomly generated tag provided to all IDs? Not sure if this would work or if the syntax is sane but:

```javascript
function id_singleton_prefix() { 
    if(global_prefix === null) {
        global_prefix = something_random;
    return global_prefix;
}

...
this.$scaled = $("<input type='checkbox' id=id_singleton_prefix() + 'scaled'>");
```